roles/network_plugin/kube-router/templates/kube-router.yml.j2

```diff
@@ -0,0 +1,223 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kube-router-cfg
+  namespace: kube-system
+  labels:
+    tier: node
+    k8s-app: kube-router
+data:
+  cni-conf.json: |
+    {
+      "name":"kubernetes",
+      "type":"bridge",
+      "bridge":"kube-bridge",
+      "isDefaultGateway":true,
+{% if kube_router_support_hairpin_mode %}
+      "hairpinMode":true,
+{% endif %}
+      "ipam": {
+        "type":"host-local"
+      }
+    }
+  kubeconfig: |
+    apiVersion: v1
+    kind: Config
+    clusterCIDR: {{ kube_pods_subnet }}
+    clusters:
+    - name: cluster
+      cluster:
+        certificate-authority: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+        server: {{ kube_apiserver_endpoint }}
```

Comment:
I'm not sure `{{ kube_apiserver_endpoint }}` covers all cases. We should look in the other template files. I think it's in the kubeadm config file.

Reply:
`grep -r server: ` output always shows `{{ kube_apiserver_endpoint }}` there
(this is what I had used as reference).

Reply:
Yes. I see. But it should be the internal endpoint for the kubernetes api, right? Not an external endpoint/loadbalanced addres etc. ? @riverzhang Do you know if this is ok? Didn't you fix the issue in the kubeadm deployment where we set the address to the api server too ?

Reply:
Fwiw it renders to local api endpoint:
```
$ kubectl get cm --namespace=kube-system kube-router-cfg -oyaml|grep server:
        server: https://127.0.0.1:6443
```

Reply:
> Fwiw it renders to local api endpoint:
> 
> ```
> $ kubectl get cm --namespace=kube-system kube-router-cfg -oyaml|grep server:
>         server: https://127.0.0.1:6443
> ```

Also on nodes ? `https://127.0.0.1:6443` is only available on masters

Reply:
I just checked. It is the correct var to use! Kube-proxy uses the same 

```
kube_apiserver_endpoint: |-
  {% if not is_kube_master and loadbalancer_apiserver_localhost -%}
       https://localhost:{{ nginx_kube_apiserver_port|default(kube_apiserver_port) }}
  {%- elif is_kube_master -%}
       https://{{ kube_apiserver_bind_address | regex_replace('0\.0\.0\.0','127.0.0.1') }}:{{ kube_apiserver_port }}
  {%- else -%}
  {%-   if loadbalancer_apiserver is defined and loadbalancer_apiserver.port is defined -%}
       https://{{ apiserver_loadbalancer_domain_name|default('lb-apiserver.kubernetes.local') }}:{{ loadbalancer_apiserver.port|default(kube_apiserver_port) }}
  {%-   else -%}
       https://{{ first_kube_master }}:{{ kube_apiserver_port }}
  {%-  endif -%}
  {%- endif %}
```

Reply:
Hi. If I'm not missing something, here is a bug:
{{ kube_apiserver_endpoint }} might has different values for different nodes. but here, all nodes used the same value, because all of them used the same ConfigMap.  
for example: for master node, it might be "https://127.0.0.1:6443", but for other nodes, it might be a LB's IP or something else.
for master node, it's okay for "https://127.0.0.1:6443", but for normal nodes, this value can't work.

Reply:
Note that `https://127.0.0.1:6443/` does work on every node (fwiw
I already have 2~3 deployments fully working using this branch), as
non master nodes get the `nginx-proxy`, as per https://github.com/kubernetes-incubator/kubespray/blob/master/docs/ha-mode.md#kube-apiserver.

Having said that, I see there'll certainly be an issue if the user chooses
to set `loadbalancer_apiserver` (external LB) instead of default
`loadbalancer_apiserver_localhost` -- much appreciate feedback on this,
Guess I could take the same approach as `contiv` does at
https://github.com/kubernetes-incubator/kubespray/blob/master/roles/network_plugin/contiv/tasks/main.yml#L23 ?